packages/scheduler/src/Scheduler.js

```diff
@@ -261,21 +302,45 @@ function unstable_scheduleCallback(priorityLevel, callback, options) {
       typeof options.timeout === 'number'
         ? options.timeout
         : timeoutForPriorityLevel(priorityLevel);
+    // if (enableProfiling) {
+    //   var _label = options.label;
+    //   if (typeof _label === 'string') {
+    //     label = _label;
+    //   }
+    // }
   } else {
     timeout = timeoutForPriorityLevel(priorityLevel);
     startTime = currentTime;
   }
 
   var expirationTime = startTime + timeout;
 
-  var newTask = {
-    id: taskIdCounter++,
-    callback,
-    priorityLevel,
-    startTime,
-    expirationTime,
-    sortIndex: -1,
-  };
+  var newTask;
+  if (enableProfiling) {
+    newTask = {
+      // __PROFILER__ only
+      isQueued: false,
+      // label: label
+
+      // !!! Keep these in sync with below branch !!!
+      id: ++taskIdCounter,
+      callback,
+      priorityLevel,
+      startTime,
+      expirationTime,
+      sortIndex: -1,
+    };
+  } else {
+    newTask = {
+      // !!! Keep these in sync with above branch !!!
+      id: ++taskIdCounter,
```

Comment:
Why is this `++taskIdCounter` now instead of `taskIdCounter++`?

Reply:
I think this is unobservable. Maybe Andrew wanted to align it with the internal counter that's currently off by one? I can revert this bit.

Reply:
Oh this is because of

```js
profilingState[CURRENT_TASK_ID] = 0;
```

I bet. I could just change the first ID to `1` though.

Reply:
Resolved